### PR TITLE
Start of the authorization system

### PR DESCRIPTION
**Pull request by bliksemman**

### CODE REVIEW

File: guides/authorization.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Authorization
```
Comment (emielvdveen) [inline guides/authorization.md:1]: Awesome doc 👍

File: guides/authorization.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+A role assignment (`Link.Authorization.RoleAssignment`) registers the assignment
+of a role to a principal on a specific entity. This is what makes having local
+roles possible.
+
+## Plug Middleware
+
+### Entity Extractor
+
+The `Link.Authorization.Plug.EntityExtractor` is used to create (bare) entities
```
Comment (emielvdveen) [inline guides/authorization.md:56]: nice

**Review by emielvdveen — APPROVED** [review]

**Review by emielvdveen — APPROVED** [review]

File: lib/green_light.ex [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  Permissions are defined by using the grant_* helpers
+  (`GreenLight.Permissions.grant_access/2`,
+  `GreenLight.Permissions.grant_actions/2`).
+
+
+  """
+  alias GreenLight.Config
+
+  @doc false
+  defmacro __using__(config) do
```
[the remark this answers is not in the document]
Comment (bliksemman) [reply lib/green_light.ex:17]: - [ ] Move more logic to separate module

File: lib/green_light.ex [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      def assign_role({:ok, entity} = result, principal, role) do
+        assign_role!(principal, entity, role)
+        result
+      end
+
+      def assign_role({:error, _} = result, _principal, _role) do
+        result
+      end
+
+      # def can?(%GreenLight.Principal{} = principal, entity, permission) do
```
Comment (bliksemman) [inline lib/green_light.ex:67]: - [ ] Remove commented code

**Review by emielvdveen — APPROVED** [review]

**Pull request merged**